frontend/src/libs/hooks/use-media-query/use-media-query.hook.ts

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useEffect, useState } from '../hooks.js';
+
+const useMediaQuery = (query: string): boolean => {
+  const [isMatched, setIsMatched] = useState(false);
+
+  const handleMediaQueryChange = useCallback(
+    (event: MediaQueryListEvent): void => {
+      setIsMatched(event.matches);
+    },
+    [],
+  );
+
+  useEffect(() => {
+    const mediaQueryList = window.matchMedia(query);
+
+    mediaQueryList.addEventListener('change', handleMediaQueryChange);
+
+    setIsMatched(mediaQueryList.matches);
```

Comment:
```suggestion
    const mediaQueryList = window.matchMedia(query);
    setIsMatched(mediaQueryList.matches);

    mediaQueryList.addEventListener('change', handleMediaQueryChange);
```

If it won't break the logic